src/main/kotlin/racingcar/RacingCarGame.kt

```diff
@@ -1,43 +1,20 @@
 package racingcar
 
-import kotlin.random.Random
+class RacingCarGame(private val trialCount: Int, private val seedMaker: SeedMaker, carCount: List<String>) {
+    private val carList = List(carCount.size) { Car(carCount[it], seedMaker) }
 
-class GameResults {
-    val gameResult
-        get() = _gameResult.toList()
-    private val _gameResult = mutableListOf<List<Int>>()
-
-    fun record(carList: List<Car>) {
-        val recordData = carList.map { it.position }
-        _gameResult.add(recordData)
-    }
-}
-
-class RacingCarGame(private val carCount: Int, private val trialCount: Int) {
-    private val defaultCarGoDecide = {
-        val randomValue = Random.nextInt(DEFAULT_RANDOM_RANGE)
-
-        randomValue >= DEFAULT_CAN_GO_THRESHOLD
-    }
-
-    private val carList = List(carCount) { Car(defaultCarGoDecide) }
-
-    private val _gameResults = GameResults()
-    fun getGameResult() = _gameResults.gameResult
+    val gameResults = GameResults()
 
     fun play() = repeat(trialCount) {
         proceed()
         record()
     }
 
+    fun getWinnerResult() = gameResults.getWinnerResult(carList)
+
     private fun record() {
-        _gameResults.record(carList)
+        gameResults.record(carList)
     }
 
     private fun proceed() = carList.forEach { it.proceed() }
```

Comment:
```suggestion
    private fun proceed(): Unit = carList.forEach { it.proceed() }
```

저는 식(expression)으로 표현하면, 항상 리턴 타입을 명확하게 적어주고 있습니다. 객체를 반환하는 함수인데, 휴먼 에러로 인해서 unit를 항상 반환을 했었던 적이 있었거든요.😂 

IDE에서 친절하게 알려주긴 하지만, 알려주기만 할 뿐 에러를 잡아주지는 않기도 해요  😭

기표님은 평소에 어떻게 쓰시나요?🤔

Reply:
저는 expression함수형은 쓰지 않는편입니다..